docs/source/advanced/backends.rst

```diff
@@ -434,3 +528,121 @@ The following tables list the functions available in the `CUDA Runtime API <http
 **Version Management**
 
 *not available*
+
+
+
+HIP
+```
+
+Current restrictions on HCC platform
+++++++++++++++++++++++++++++++++++++
+
+- Workaround for unsupported ``syncthreads_{count|and|or}``.
+  - uses temporary shared value and atomics
+- Workaround for buggy ``hipStreamQuery``, ``hipStreamSynchronize``.
+  - introduces own queue management
+  - ``hipStreamQuery`` and ``hipStreamSynchronize`` did not work in multithreaded environment
+- Workaround for missing ``cuStreamWaitValue32``.
+  - polls value each 10ms
+- device constant memory not supported yet
+- note, that ``printf`` in kernels is still not supported in HIP
+- exclude ``hipMalloc3D`` and ``hipMallocPitch`` when size is zero otherwise they throw an Unknown Error
+- ``TestAccs`` excludes 3D specialization of Hip back-end for now because ``verifyBytesSet`` fails in ``memView`` for 3D specialization
+- ``dim3`` structure is not available on device (use ``alpaka::vec::Vec`` instead)
+- Constructors' attributes unified with destructors'.
+  - host/device signature must match in HIP(HCC)
```

Comment:
I do not understand this point.

Reply:
I don't know. Matthias wrote this but I think, the HIP section need a general overwork, because some parts, like the `hipcc` (which is deprecated) are outdated.

Reply:
I opened a new Issue, which contains checking the HIP backend documentation: #1105